### PR TITLE
Add new target `operations`/`ops` 

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it

This PR adds a new target `operations`/`ops`

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
Add the new target `operations`
```

### CODE REVIEW

File: pkg/command/root.go [hunk 1]
```diff
@@ -297,6 +299,18 @@ var tmcCmd = &cobra.Command{
 	},
 }
 
+var opsCmd = &cobra.Command{
+	Use:     "operations",
+	Short:   "Tanzu CLI Operations related commands",
```
Comment (anujc25) [inline pkg/command/root.go:304]: @vuil any suggestion for the help for this target?

File: go.mod [hunk 2]
```diff
@@ -6,6 +6,8 @@ replace cloud.google.com/go => cloud.google.com/go v0.102.1
 
 replace github.com/vmware-tanzu/tanzu-cli/test/e2e/framework => ./test/e2e/framework
 
+replace github.com/vmware-tanzu/tanzu-plugin-runtime => github.com/anujc25/tanzu-plugin-runtime v0.0.0-20240214002508-cc83ffc09fda
```
Comment (anujc25) [inline go.mod:9]: Update once https://github.com/vmware-tanzu/tanzu-plugin-runtime/pull/163 is merged.

**Comment by marckhouzam** [comment]
While I continue reviewing, I thought I'd mention that the shell completion logic was not updated for the new `operations/ops` target.  That change starts with `pkg/command/completion_helper.go`, but needs to be added to every completion function that suggests a target;  you should be able to find them quickly by looking for `compTMCTarget`.

The unit tests will also need to be updated to match... Sorry about that.

**Review by vuil — APPROVED** [review]
lgtm, a reminder to update go.mod before merging.

**Review by prkalle — APPROVED** [review]
LGTM. Thanks for the changes!

File: pkg/command/root.go (hunk 1 above)
Comment (anujc25) [reply pkg/command/root.go]: Updated.

**Review by marckhouzam — COMMENTED** [review]
We need to update:
1. `TargetList` in `pkg/common/constants.go`
2. `TargetList` in `test/e2e/framework/framework_constants.go`

What about documentation?  Should we add something in `docs/quickstart/quickstart.md` and/or `docs/full/README.md`?

When there are not `ops` plugins installed we get
```
$ tz ops                                                                                                                         <gke_dev-adhol>
Note: No plugins are currently installed for the "operations" target.
      Such plugins will be accessible when a "operations" context is created/activated or a "operations" plugin is explicitly installed.
...
```
should we change the note since we now have a target that cannot be a context?  Maybe just for this target?

As a follow-up PR, we may want to consider adding `operations` plugins to the test central repo.

**Comment by anujc25** [comment]
> should we change the note since we now have a target that cannot be a context? Maybe just for this target?
> 
Can we remove this line completely? `Such plugins will be accessible when a "<target>" context is created/activated or a "<target>" plugin is explicitly installed.` 


> As a follow-up PR, we may want to consider adding `operations` plugins to the test central repo.
>
👍

**Comment by anujc25** [comment]
> Such plugins will be accessible when a "<target>" context is created/activated or a "<target>" plugin is explicitly installed.

I have removed `Such plugins will be accessible when a "<target>" context is created/activated or a "<target>" plugin is explicitly installed.` help line for now. If we think we need to add it back, I can add it back as part of the followup PR.

**Pull request merged**